### ISSUE TITLE
Wrong method being written out in docs from programgen

### ISSUE THREAD

**Issue by pierskarsenbarg** [pulumi/pulumi#11555]
### What happened?

I don't have a complete list, but there are a couple of occasions where the Azure Classic provider docs reference the `getclientconfig` method. In the C# examples, it says to use the following method: `Azure.Core.GetClientConfig.Invoke();` but that doesn't exist. It should be `Azure.Core.GetClientConfig.InvokeAsync();`.

### Steps to reproduce

https://www.pulumi.com/registry/packages/azure/api-docs/core/getclientconfig/#example-usage
https://www.pulumi.com/registry/packages/azure/api-docs/compute/diskencryptionset/#example-usage 

(both C#)

### Expected Behavior

Method should be `Azure.Core.GetClientConfig.InvokeAsync();`

### Actual Behavior

It shows as: `Azure.Core.GetClientConfig.Invoke();`

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
This looks like a .NET programgen bug.

**Comment by Zaid-Ajaj** [comment]
@justinvp Yeah the current program-gen defaults to using output-versiined invokes everywhere but it looks like we don't generate output-versioned invokes for functions that don't have inputs 🧐 it makes sense because we don't _need_ it but the alternatives here are both ugly:
  - having to await the invoke async call means that program-needs to handle these cases separately
  - having to call Output.Create(InvokeAsync()) which is what we avoided to generate in the SDKs in the first place

Feels like adding an output-versioned invoke to the C# sdks would make for much nicer and consistent program-gen
